### PR TITLE
Modal functionality

### PR DESCRIPTION
**Pull request by kevin-quach**
https://trello.com/c/r1BpwMIr

**Changes:**

- [x] Added Modal JS,HTML,SCSS
- [x] Updated style.scss to include new dependencies
- [x] Modal functionality built with 3 size options
  

**How to test:**
- Define a trigger for the modal with `data-modal="#myModal"`
- Create modal based on the structure outlined in the wiki [here](https://github.com/sio-digital-team/web-ui-sdk/wiki/Components#modal).
- See if modal opens 
- Modal closes when clicking on the backdrop or any element that has the `data-close` attribute.
  

**Todo:**
- [x] Modal documentation

### CODE REVIEW

File: public/js/modal.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  /* Simple Modal */
+  $.fn.modal = function(opt) {
+    return this.each(function() {
+      //Defaults
+      var defaults = {
+        speed: 250,
+        complete: function() {}
+      };
+
+      console.info("MAKING BACKDROP");
```
Comment (Magitron) [inline public/js/modal.js]: Just wondering, do we need this log?

File: public/js/modal.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+$(document).ready(function() {
+  $('[data-modal]').modal();
+});
+
+
```
Comment (Magitron) [inline public/js/modal.js]: I think we can remove some spaces here. No big deal though.

**Comment by Magitron** [comment]
tenor-20843510

File: public/js/modal.js (hunk 1 above)
Comment (kevin-quach) [reply public/js/modal.js]: When you forget to remove your test consoles :|

**Pull request merged**